src/rgw/driver/rados/rgw_notify.cc

```diff
@@ -477,9 +475,10 @@ class Manager : public DoutPrefixProvider {
         boost::asio::spawn(yield, std::allocator_arg, make_stack_allocator(),
           [this, &notifs_persistency_tracker, &queue_name, entry_idx,
            total_entries, &end_marker, &remove_entries, &has_error,
-           token = waiter.make_token(), &entry, &needs_migration_vector,
+           &tw, &entry, &needs_migration_vector,
            push_endpoint = push_endpoint.get(),
            &topic_info](boost::asio::yield_context yield) {
+            tokens_waiter::token token(tw);
```

Comment:
this doesn't create the token until we're inside the coroutine. doesn't this leave a window where we haven't incremented `pending_tokens` yet, so a racing `async_wait()` call would return without waiting?